File: examples/python/campaign/target_url.py
from dataclasses import dataclass


@dataclass(frozen=True)
class TargetURL:

    _value: str

    def __post_init__(self) -> None:
        if not (self._value.startswith("http://") or self._value.startswith("https://")):
            raise ValueError(
                f"invalid target URL {self._value!r}: must start with "
                "http:// or https://"
            )

    def __str__(self) -> str:
        return self._value

File: examples/running/campaign/target_url.go
package campaign

import (
	"fmt"
	"strings"
)

type TargetURL struct {
	value string
}

func NewTargetURL(value string) (TargetURL, error) {
	if !strings.HasPrefix(value, "http://") && !strings.HasPrefix(value, "https://") {
		return TargetURL{}, fmt.Errorf("invalid target URL %q: must start with http:// or https://", value)
	}
	return TargetURL{value: value}, nil
}

func MustNewTargetURL(value string) TargetURL {
	u, err := NewTargetURL(value)
	if err != nil {
		panic(err)
	}
	return u
}

func (u TargetURL) String() string {
	return u.value
}
